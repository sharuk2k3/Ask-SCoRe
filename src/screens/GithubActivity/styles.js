import {StyleSheet, Platform, Dimensions} from "react-native";

let deviceWidth = Dimensions.get("window").width;
let deviceHeight = Dimensions.get("window").height;

export default StyleSheet.create({
  SafeAreaStyle: {backgroundColor: "#e6e6e6"},

  ViewStyle: {marginBottom: 15},

  ViewStyle1: {
    width: "90%",
    alignSelf: "center",
    height: deviceHeight * 0.15,
    backgroundColor: "white",
    marginBottom: 20,
    borderRadius: 7,
    borderBottomRightRadius: 20,
    borderTopRightRadius: 20,
    flexDirection: "row",
    flex: 1,
  },

  TextStyle: {
    fontSize: 17,
    fontWeight: Platform.OS === "ios" ? "400" : "normal",
    marginHorizontal: 5,
    marginTop: 2,
  },

  TextStyle1: {
    fontSize: 11,
    fontWeight: "300",
    marginHorizontal: 5,
    maxHeight: 40,
    maxWidth: 260,
  },

  TextStyle2: {
    color: "#23598B",
    fontSize: 12,
    fontWeight: "normal",
    marginHorizontal: 5,
    maxHeight: 35,
    marginTop: 5,
  },

  ViewStyle2: {alignSelf: "flex-end", flex: 2.6},

  ViewStyle3: {flexDirection: "row"},

  DividerView: {
    borderBottomColor: "#d4d4d4",
    borderBottomWidth: 1,
    marginBottom: 3,
    marginHorizontal: "2%",
    marginVertical: 3,
  },

  ViewStyle4: {
    borderBottomColor: "black",
    borderRightWidth: 2,
  },

  ImageStyle: {
    width: deviceWidth * 0.235,
    height: deviceHeight * 0.15,
    alignSelf: "flex-end",
    borderBottomRightRadius: 20,
    borderTopRightRadius: 20,
  },

  ViewStyle5: {flex: 7.4, padding: 5, marginLeft: 5},
});
