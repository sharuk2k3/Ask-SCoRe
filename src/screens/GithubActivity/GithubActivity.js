import axios from "axios";
import React, {Component} from "react";
import {
  View,
  Text,
  FlatList,
  SafeAreaView,
  Image,
  TouchableOpacity,
  Linking,
} from "react-native";
import styles from "./styles";
import {GITHUB_API} from "../../config/config";

class GithubActivity extends Component {
  state = {githubData: []};

  componentDidMount() {
    const {githubData} = this.state;
    axios
      .get(GITHUB_API)
      .then(response => {
        this.setState({githubData: response.data});
      })
      .catch(error => {
        alert(error);
      });
  }

  render() {
    const {githubData} = this.state;
    return (
      <SafeAreaView style={styles.SafeAreaStyle}>
        <View>
          <View style={styles.ViewStyle} />
          <FlatList
            data={githubData}
            renderItem={({item}) => {
              return (
                <View>
                  <TouchableOpacity
                    onPress={() => Linking.openURL(item.html_url)}>
                    <View style={styles.ViewStyle1}>
                      <View style={styles.ViewStyle5}>
                        <Text style={styles.TextStyle}>{item.name}</Text>
                        <View style={styles.DividerView} />
                        <Text style={styles.TextStyle1}>
                          {item.description}
                        </Text>
                        <Text style={styles.TextStyle2}>{item.language}</Text>
                      </View>
                      <View style={styles.ViewStyle2}>
                        <View style={styles.ViewStyle3}>
                          <Image
                            source={{uri: item.owner.avatar_url}}
                            style={styles.ImageStyle}
                          />
                        </View>
                      </View>
                    </View>
                  </TouchableOpacity>
                </View>
              );
            }}
            keyExtractor={(item, index) => "key" + index}
            ListFooterComponent={<View style={{height: 50}} />}
          />
        </View>
      </SafeAreaView>
    );
  }
}
export default GithubActivity;
