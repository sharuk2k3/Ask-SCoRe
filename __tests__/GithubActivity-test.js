import "react-native";
import React from "react";
import GithubActivity from "../src/screens/GithubActivity/GithubActivity";
import renderer from "react-test-renderer";

jest.useFakeTimers();

test("renders Notification Screen correctly", async () => {
  const tree = renderer.create(<GithubActivity />).toJSON();
  expect(tree).toMatchSnapshot();
});
