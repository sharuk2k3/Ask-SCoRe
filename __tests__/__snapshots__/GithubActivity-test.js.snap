// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders Notification Screen correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Object {
      "backgroundColor": "#e6e6e6",
    }
  }
>
  <View>
    <View
      style={
        Object {
          "marginBottom": 15,
        }
      }
    />
    <RCTScrollView
      ListFooterComponent={
        <View
          style={
            Object {
              "height": 50,
            }
          }
        />
      }
      data={Array []}
      disableVirtualization={false}
      getItem={[Function]}
      getItemCount={[Function]}
      horizontal={false}
      initialNumToRender={10}
      keyExtractor={[Function]}
      maxToRenderPerBatch={10}
      onContentSizeChange={[Function]}
      onEndReachedThreshold={2}
      onLayout={[Function]}
      onMomentumScrollEnd={[Function]}
      onScroll={[Function]}
      onScrollBeginDrag={[Function]}
      onScrollEndDrag={[Function]}
      removeClippedSubviews={false}
      renderItem={[Function]}
      scrollEventThrottle={50}
      stickyHeaderIndices={Array []}
      updateCellsBatchingPeriod={50}
      viewabilityConfigCallbackPairs={Array []}
      windowSize={21}
    >
      <View>
        <View
          onLayout={[Function]}
        >
          <View
            style={
              Object {
                "height": 50,
              }
            }
          />
        </View>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
